### PR TITLE
Ensure synthesized BoundBaseReference always uses correct type.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 1]
```diff
@@ -58150,5 +58150,256 @@ public interface ITest33
...
+    {
+        I1.M1();
+    }
+}
+";
+
+            var compilation0 = CreateCompilation(source0, options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
+            compilation0.VerifyDiagnostics();
+
+            var verifier = CompileAndVerify(compilation0, expectedOutput: !ExecutionConditionUtil.IsMonoOrCoreClr ? null : @"M1", verify: VerifyOnMonoOrCoreClr);
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: >!ExecutionConditionUtil.IsMonoOrCoreClr ? null : @"M1" [](start = 74, length = 54)

Consider inverting this conditional to make it simpler to read.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: > Consider inverting this conditional to make it simpler to read.

I find it  much easier to have the string at the end, especially when it is multi-line.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a very small change

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a very small change

**Pull request merged**